spec/dummy/app/resources/article_resource.rb

```diff
@@ -24,4 +24,8 @@ def id=(external_id)
   # Setting this attribute is an easy way to test that authorizations work even
   # when the model has validation errors
   attributes :blank_value
+
+  def self.creatable_fields(context)
+    super + [:id]
+  end
```

Comment:
I launched our test suite on our CI with jsonapi-resource 0.9 and this PR and we get a bunch of errors (unexpected `406 - Not Acceptable` responses). It's hard to tell right now if it's because of upgrading to JR 0.9 or this PR specifically. 
Sadly I didn't have much time to investigate more this week. I'll try to block an hour next week.